# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: devtools/client/webconsole/test/browser/browser_webconsole_trackingprotection_errors.js
# Commit: ff5ea82f6090
# Full Hash: ff5ea82f6090a4a1645a1a32749a4ae9102f015b
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2025-04-17 16:30:04
# Regressor Bug: 1957355
# File Overlap Count: 1
# Description:
#   Bug 1957355 - No storage access for resources injected by content-scripts, r=timhuang,necko-reviewers,devtools-reviewers,anti-tracking-reviewers,nchevobbe,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D244204
# ==============================================================================

diff -r 04eed8759454 -r ff5ea82f6090 devtools/client/webconsole/test/browser/browser_webconsole_trackingprotection_errors.js
--- a/devtools/client/webconsole/test/browser/browser_webconsole_trackingprotection_errors.js	Thu Apr 17 10:56:59 2025 +0000
+++ b/devtools/client/webconsole/test/browser/browser_webconsole_trackingprotection_errors.js	Thu Apr 17 10:56:59 2025 +0000
@@ -238,6 +238,81 @@
   Services.perms.removeFromPrincipal(p, "cookie");
 });
 
+add_task(async function testCookieBlockedForUserContentResourceMessage() {
+  info("Test usercontent resource cookie blocking");
+  // Bug 1518138: GC heuristics are broken for this test, so that the test
+  // ends up running out of memory. Try to work-around the problem by GCing
+  // before the test begins.
+  Cu.forceShrinkingGC();
+  await pushPref("privacy.antitracking.isolateContentScriptResources", true);
+
+  const extension = ExtensionTestUtils.loadExtension({
+    manifest: {
+      content_scripts: [
+        {
+          matches: [`*://example.com/*/test-blank.html`],
+          js: ["content_scripts.js"],
+          all_frames: true,
+        },
+      ],
+    },
+    files: {
+      "content_scripts.js": `
+            // An image
+            const img = document.createElement("img");
+            img.src = "https://example.com/${TEST_PATH}test-image.png";
+            document.body.appendChild(img);
+
+            // A script
+            const script = document.createElement("script");
+            script.src = "https://example.com/${TEST_PATH}empty-with-cookie.js";
+            document.body.appendChild(script);
+
+            // An iframe
+            const iframe = document.createElement("iframe");
+            iframe.src = "https://example.com/${TEST_PATH}test-blank-with-cookie.html";
+            document.body.appendChild(iframe);
+          `,
+    },
+  });
+
+  await extension.startup();
+
+  const { hud, win } = await openNewWindowAndConsole(
+    "https://example.com/" + TEST_PATH + "test-blank.html"
+  );
+
+  await waitFor(
+    () =>
+      findWarningMessage(
+        hud,
+        "Request to access cookie or storage on “https://example.com/" +
+          `${TEST_PATH}test-image.png” was blocked because we are blocking all ` +
+          "storage access requests."
+      ) &&
+      findWarningMessage(
+        hud,
+        "Request to access cookie or storage on “https://example.com/" +
+          `${TEST_PATH}empty-with-cookie.js” was blocked because we are ` +
+          "blocking all storage access requests."
+      ) &&
+      findWarningMessage(
+        hud,
+        "Request to access cookie or storage on “https://example.com/" +
+          `${TEST_PATH}test-blank-with-cookie.html” was blocked because ` +
+          "we are blocking all storage access requests."
+      )
+  );
+  ok(true, "Third-party storage access blocked message was displayed");
+
+  // We explicitely destroy the toolbox in order to ensure waiting for its full destruction
+  // and avoid leak / pending requests
+  await hud.toolbox.destroy();
+  win.close();
+
+  await extension.unload();
+});
+
 function getStorageErrorUrl(category) {
   const BASE_STORAGE_ERROR_URL =
     "https://developer.mozilla.org/docs/Web/Privacy/Guides/Storage_Access_Policy/Errors/";